completion.py

```diff
@@ -44,7 +47,30 @@ def get_completion(clones_dir: str, repo: str) -> tuple[float, 'TranslatorsData'
             hide_reserved=False,
             api_url='',
         ).completion
-    return completion, translators_data, branch
+
+    try:
+        # Fetch commit from before 30 days ago and checkout
+        commit = subprocess.check_output(
+            ['git', 'rev-list', '-n', '1', '--before="30 days ago"', 'HEAD'],
+            cwd=clone_path,
+            text=True,
+        ).strip()
+        subprocess.run(['git', 'checkout', commit], cwd=clone_path, check=True)
```

Comment:
1. Please convert to gitpython, which is used around already.

```suggestion
        repo = git.Repo(clone_path)
        commit = next(repo.iter_commits('HEAD', max_count=1, before="30 days ago"))
        repo.git.checkout(commit.hexsha)
```

2. Please also checkout the CPython repo at ealier revision and `make gettext`. This way we will also show regressions (negative changes), when the source is updated, but translation is not.

Reply:
I realised 2 will be suboptimal without a refactor, because we would checkout and rebuild way too many times.

I already planned to make the language clones less volatile. Therefore I'd be ok to merge for now without 2.